proposals/fixed-sized-buffers.md

```diff
@@ -0,0 +1,72 @@
+# Fixed Sized Buffers
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Provide a general-purpose and safe mechanism for declaring fixed sized buffers to the C# language.
+
+## Motivation
+[motivation]: #motivation
+
+Today, users have the ability to create fixed-sized buffers in an unsafe-context. However, this requires the user to deal with pointers, manually perform bounds checks, and only supports a limited set of types (`bool`, `byte`, `char`, `short`, `int`, `long`, `sbyte`, `ushort`, `uint`, `ulong`, `float`, and `double`).
+
+With a few minor tweaks, we could provide general-purpose fixed-sized buffers which support any type, can be used in a safe context, and have automatic bounds checking performed.
+
+## Detailed design
+[design]: #detailed-design
+
+One would declare a safe fixed-sized buffer via the following:
+
+```C#
+public fixed DXGI_RGB GammaCurve[1025];
+```
+
+The declaration would get translated into an internal representation by the compiler that is similar to the following
+
+```C#
+[CompilerGenerated, StructLayout(LayoutKind.Sequential, Pack = 1)]
+struct <GammaCurve>e__StaticDelegate
```

Comment:
Will Fix. Cloned one of my other proposals as a template and apparently missed this part :)